src/window.rs

```diff
@@ -0,0 +1,149 @@
+
+    let mut canvas = window.into_canvas().build().expect("Couldn't create CanvasBuilder");
+    let texture_creator = canvas.texture_creator();
+
+    let mut texture = texture_creator.create_texture_from_surface(surface_img)
+        .expect("couldn't create texture from surface");
+
+    // calculates new location for surface from window origin so that
+    // the image is centered in the window
+    let center_x = ((window_width - output_image_width) as f32 / 2.0_f32) as i32;
+    let center_y = ((window_height - output_image_height) as f32 / 2.0_f32) as i32; 
+    
+    // makes background white
+    canvas.set_draw_color(Color::RGB(255, 255, 255));
+    canvas.clear();
+
+    // displays image in the window
+    canvas.copy(
+        &texture, 
+        None, 
+        Rect::new(center_x, center_y, output_image_width, output_image_height))
+        .unwrap();
+    canvas.present();
+
+    // create and start events loop to keep window open until Esc 
+    let mut event_pump = sdl.event_pump().unwrap();
+    event_pump.enable_event(sdl2::event::EventType::Window);
+    'running: loop {
+        for event in event_pump.poll_iter() {
+            match event {
+                Event::Quit {..} |
+                    Event::KeyDown {keycode: Some(Keycode::Escape), .. } => {
+                        break 'running
+                    },
+                Event::Window { win_event: WindowEvent::Resized(x, y), .. } => {
+                    
+                    let x = x as u32;
+                    let y = y as u32;
+                    // resize image if necessary to fit into the window
+                    let (output_image_width, output_image_height, output_image) = 
+                        if image_height < y && image_width < x {
+                            (image_width, image_height, image.clone())
+                        } else {
+                            let scale = {
+                                let width_scale = x as f32 / image_width as f32; 
+                                let height_scale = y as f32 / image_height as f32;
+                                if  width_scale < height_scale {
+                                    width_scale
+                                } else {
+                                    height_scale
+                                }
+                            };
+                            let height = (scale * image_height as f32) as u32;
+                            let width = (scale * image_width as f32) as u32;
+                            let output_image = image::imageops::resize(
+                                image, 
+                                width, 
+                                height, 
+                                image::FilterType::Triangle);
+                            (width, height, output_image)
+                        };
+                    
+
+                    let pitch  = output_image_width * CHANNEL_COUNT;
```

Comment:
This next section also appears to be copy-pasted from above, so should be moved into function to be called from both locations.